### PR TITLE
Fix: Breadcrumb should work with react-router@4

### PR DESCRIPTION
**Pull request by benjycui**
Close: https://github.com/ant-design/ant-design/issues/5644

### CODE REVIEW

File: components/breadcrumb/demo/router-4.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+---
+order: 3
+iframe: 200
+reactRouter: react-router-dom
```
Comment (benjycui) [inline components/breadcrumb/demo/router-4.md]: bisheng-plugin-antd 已经更新，可以在 react-router react-router-dom 中二选一。

Comment (afc163) [inline components/breadcrumb/demo/router-4.md]: react-router-dom => react-router@4?

react-router =>  react-router@3?

Comment (benjycui) [inline components/breadcrumb/demo/router-4.md]: Actually, react-router@4 and react-router-dom@4 are different.

And we should use react-router-dom in browser.

Comment (afc163) [inline components/breadcrumb/demo/router-4.md]: https://github.com/ant-design/bisheng-plugin-antd/commit/3649f2034027ab2a7539bbef86b3c419fb36dce1#diff-94329ac063013015d4f4d7d7bfc79de3R129 make me confused, how about:

```
- insertedScript: react-router@3.x
- insertedScript: react-router-dom@4.x
```

**Comment by afc163** [comment]
```
Summary of all failing tests

 FAIL  components/breadcrumb/__tests__/demo.test.js

  ● renders ./components/breadcrumb/demo/router-4.md correctly

    Cannot find module 'react-router-dom' from 'router-4.md'

      

      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:169:17)

      at Object.<anonymous> (components/breadcrumb/demo/router-4.md:19:23)
```

**Comment by teophilus** [comment]
looking forward to this PR merge 👍

**Comment by afc163** [comment]
ping~

**Comment by newswim** [comment]
I'd love to see this merged. There appears to be a failing test,

```
Summary of all failing tests
 FAIL  components/breadcrumb/__tests__/demo.test.js
  ● renders ./components/breadcrumb/demo/router-4.md correctly
    Cannot find module 'react-router-dom' from 'router-4.md'
      
      at Resolver.resolveModule (node_modules/jest-resolve/build/index.js:169:17)
      at Object.<anonymous> (components/breadcrumb/demo/router-4.md:19:23)
```

Looks like you might need to add `react-router` and `react-router-dom` to the branch.

**Comment by afc163** [comment]
```
Summary of all failing tests

 FAIL  components/breadcrumb/__tests__/demo.test.js (7.052s)

  ● renders ./components/breadcrumb/demo/router-4.md correctly

    Invariant Violation: Hash history needs a DOM

      

      at invariant (node_modules/invariant/invariant.js:42:15)

      at createHashHistory (node_modules/react-router-dom/node_modules/history/createHashHistory.js:69:27)

      at new HashRouter (node_modules/react-router-dom/HashRouter.js:42:195)

      at node_modules/react-dom/lib/ReactCompositeComponent.js:294:18

      at measureLifeCyclePerf (node_modules/react-dom/lib/ReactCompositeComponent.js:75:12)

      at ReactCompositeComponentWrapper._constructComponentWithoutOwner (node_modules/react-dom/lib/ReactCompositeComponent.js:293:16)

      at ReactCompositeComponentWrapper._constructComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:279:21)

      at ReactCompositeComponentWrapper.mountComponent (node_modules/react-dom/lib/ReactCompositeComponent.js:187:21)

      at Object.mountComponent (node_modules/react-dom/lib/ReactReconciler.js:45:35)

      at node_modules/react-dom/lib/ReactServerRendering.js:45:36
```

**Comment by newswim** [comment]
If Jest is being used to test, the environment might need to be set to `jsdom` --- luckily, there's a new shorthand for that:

```js
/**
 * @jest-environment jsdom
 */

import React from 'react'
/* test code */
```

I've had pretty good luck so far switching between jsdom and node, depending on the component's needs.

**Comment by yesmeck** [comment]
Because no DOM when doing SSE, skip demo test here https://github.com/ant-design/ant-design/blob/master/components/breadcrumb/__tests__/demo.test.js#L5

**Comment by benjycui** [comment]
Actually, this PR is not finished...

**Comment by benjycui** [comment]
I will handle this .

**Comment by gRoberts84** [comment]
Is there any update on this?

**Comment by benjycui** [comment]
@gRoberts84 https://ant.design/components/breadcrumb/#components-breadcrumb-demo-router-4

**Comment by fgiarritiello** [comment]
How does this work for routes such as this /user/edit/:id ? The example provided does not handle this case.

Tnx

**Pull request merged**